Update new-plans.md callouts for Dec. 10th date

sarahsanders-docker opened the pull request:
## Description
- Marketing team will be updating the website and pricing pages to reflect the Nov. 15th to Dec. 10th date change for self service consolidated plans, so we need to update the callouts in our current docs to reflect this
- Updated the `new-plans.md` include to reflect the date changes for self-serve new plans, as well as the Feb. 28 consumption dates

## Related issues or tickets
[ENGDOCS-2288](https://docker.atlassian.net/browse/ENGDOCS-2303?atlOrigin=eyJpIjoiYzQ4ZWM4ZjkyOTNlNDE1MmIxYzRkN2FhMmViNWFmODgiLCJwIjoiaiJ9)

## Reviews
@docker/billing

- [ ] Technical review
- [ ] Editorial review
- [ ] Product review

[ENGDOCS-2288]: https://docker.atlassian.net/browse/ENGDOCS-2288?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

craig-osterhout commented:
I added another note that didn't use the includes that should be updated at https://github.com/docker/docs/blob/main/content/manuals/docker-hub/download-rate-limit.md?plain=1#L147

sarahsanders-docker commented:
> I added another note that didn't use the includes that should be updated at https://github.com/docker/docs/blob/main/content/manuals/docker-hub/download-rate-limit.md?plain=1#L147

Thank you! I just updated that one too

craig-osterhout reviewed (approved):
LGTM.
Only nit is the "before February 28" made me second guess if it starts counting on the 28th or the next day, but it's clarified in the more details link.
